gapic-generator/lib/google/gapic/schema/loader.rb

```diff
@@ -16,6 +16,10 @@
 
 require "google/gapic/schema/wrappers"
 
+$LOAD_PATH.unshift ::File.expand_path("../../../../../vendor", __FILE__)
```

Comment:
Would `relative_require` be better than a `$LOAD_PATH` change? Why are these files being vendored?

Reply:
I don't really like having this vendored directory. Do we have any alternatives?

If we must do it I would prefer to have the underlying protos added here as a submodule with a build step to compile them. I know they don't change much, but it will be difficult to know when things get out of date how they were compiled, etc. if we just copy them over.

Reply:
The only alternative that comes to mind would be to load the protos rather than pre-compiling them but ruby-protobuf doesn't support loading protos at runtime that I know of.

Reply:
I added the protos to the repo and added rake tasks to compile the protos into lib and removed vendor entirely.

WDYT?

Reply:
Could we `protoc` them as part of a build step and exclude the generated Ruby files from source control?

Reply:
Ah, I was too slow. Yes, I like that approach better but shouldn't we exclude the generated `.pb.rb` files?

I would also prefer to add the protos as a submodule dependency so it's easier to track them, but we can always do that later.

Reply:
That seems reasonable, I'd like some guidance from @quartzmo or @blowmage on the best way to do that.

Reply:
@jbolinger You mean publishing the generated `.pb.rb` files in another GitHub repo?

Reply:
Oh nm, I see the protos now. Yes, if those are already published I can see that it might work great to add that repository as a submodule, rather than to copy them.

Reply:
@quartzmo no, I mean excluding any generated files from source control entirely and adding https://github.com/googleapis/api-common-protos as a git submodule to this project so that there is no duplication.

Reply:
I'm gonna give this a shot using `Gem#<post|pre>_install`.

Reply:
@landrito The build for this repo would use the protos included via git submodule to produce the `.pb.rb` files and include the `.pb.rb` files in the gem for this repo. So what is `Gem#<post|pre>_install` for? Sorry don't follow.

Reply:
So the `.pb.rb` files would not be distributed in the gem for this repo?

Reply:
I would add a rake task to generate the files and include that task in the default build task. Then I would exclude those files by adding `/lib/**/*.pb.rb` to the `.gitignore` file, but include them in the gem by also adding `Dir.glob("lib/**/*.pb.rb")` to the file list in the `gapic-generator.gemspec` file.

Reply:
The .pb.rb files will be distributed in the gem but not in the github repo.

Ah that makes sense @blowmage I'll do that.

Reply:
What @blowmage says is exactly what I was thinking too.

Reply:
I'd prefer to use the submodule approach so that we don't take on a dependency to autosynth. It's a more common practice anyway IMHO.

Reply:
Yep, just an option. One nice thing about committing generated files is that you know what changes happened when. Happy with whatever you settle on.